tests/trace/test_iterators.py

```diff
@@ -0,0 +1,142 @@
+import threading
+
+import pytest
+
+from weave.utils.iterators import ThreadSafeInMemorySequence
+
+
+def test_basic_sequence_operations():
+    # Test basic sequence operations
+    iterator = ThreadSafeInMemorySequence(iter(range(10)))
+    assert len(iterator) == 10
+    assert iterator[0] == 0
+    assert iterator[1:3] == [1, 2]
+    assert list(iterator) == list(range(10))
+
+
+def test_empty_iterator():
+    # Test behavior with empty iterator
+    iterator = ThreadSafeInMemorySequence(iter([]))
+    assert len(iterator) == 0
+    with pytest.raises(IndexError):
+        _ = iterator[0]
+    assert list(iterator) == []
```

Comment:
you can test these, but they are also tested "for free" with the `Sequence` typecheck